File: app/src/main/java/com/alamodrafthouse/injection/module/ActivityModule.java
package com.alamodrafthouse.injection.module;

import com.alamodrafthouse.injection.scope.ActivityScope;
import com.alamodrafthouse.ui.category.CategoryAdapter;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Sekhar on 4/8/15.
 */
@Module public class ActivityModule {

  @Provides @ActivityScope public CategoryAdapter provideCategoryAdapter() {
    return new CategoryAdapter();
  }
}

File: app/src/main/java/com/mutualmobile/praxis/injection/module/ActivityModule.kt
package com.mutualmobile.praxis.injection.module

import android.app.Activity
import com.mutualmobile.praxis.injection.scope.ActivityScope
import dagger.Module
import dagger.Provides
import dagger.android.support.DaggerAppCompatActivity
import java.util.*

@Module
(includes = arrayOf(ActivityCommonModule::class))
class ActivityModule {
  @Provides
  fun provideUUID() = UUID.randomUUID().toString()

  @Provides
  @ActivityScope
  fun provideActivity(activity: DaggerAppCompatActivity):
      Activity = activity
}